test/wallet-rpc-test.js

```diff
@@ -0,0 +1,317 @@
+      }
+
+      {
+        // Create a wallet that manages the private keys itself
+        const response = await wclient.createWallet(this.standardWalletId);
+        assert.equal(response.id, this.standardWalletId);
+
+        const info = await wclient.getAccount(this.standardWalletId, 'default');
+        assert.equal(info.watchOnly, false);
+      };
+    });
+
+    // The rpc interface requires the wallet to be selected first
+    // the "standardWalletId" is not watch only, and the
+    // "watchOnlyWalletId" is watch only. Select both and assert that
+    // the watch only property that is returned properly matches
+    it('should return iswatchonly correctly', async () => {
+      const receive = toAddress(this.xpub, {
+        network,
+        branch: 0,
+        index: 0,
+        enc: 'string'
+      });
+
+      {
+        await wclient.execute('selectwallet', [this.standardWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.iswatchonly, false);
+      }
+      {
+        await wclient.execute('selectwallet', [this.watchOnlyWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.iswatchonly, true);
+      }
+    });
+
+    // Assert that getaddressinfo returns the same address.
+    // The gotcha would be returning an address with the incorrect
+    // hrp which would be a problem with the network the address
+    // object uses
+    it('should return the correct address', async () => {
+      const receive = toAddress(this.xpub, {
+        network,
+        branch: 0,
+        index: 0,
+        enc: 'string'
+      });
+
+      await wclient.execute('selectwallet', [this.watchOnlyWalletId]);
+      const response = await wclient.execute('getaddressinfo', [receive]);
+      assert.equal(response.address, receive);
+    });
+
+    // Select the watch only wallet that was created with the xpub
+    // at account index 0', derive the same address locally
+    // and pass to getaddressinfo, assert that ismine is true,
+    // select the standard wallet that has a different hd seed
+    // and then pass the same address to getaddressinfo and
+    // assert that ismine is false
+    it('should detect owned address', async () => {
+      const receive = toAddress(this.xpub, {
+        network,
+        branch: 0,
+        index: 0,
+        enc: 'string'
+      });
+
+      {
+        await wclient.execute('selectwallet', [this.watchOnlyWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.ismine, true);
+      }
+      {
+        await wclient.execute('selectwallet', [this.standardWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.ismine, false);
+      }
+    });
+
+    it('should detect a p2sh address', async () => {
+      const pubkeys = [
```

Comment:
Using a well known test vector is probably a good idea here. Will update to use vectors where possible